### PR TITLE
show side navigation for foreign profiles and filter navigation entries

### PR DESCRIPTION
**Pull request by Tschuck**
- [CORE-1158]

[CORE-1158]: https://evannetwork.atlassian.net/browse/CORE-1158

### CODE REVIEW

File: dapps/profile.vue/src/components/root/root.ts [hunk 1]
```diff
@@ -72,6 +78,16 @@ export default class ProfileRootComponent extends mixins(EvanComponent) {
...
   }
 
+  /**
+   * Check if a route name is accessable by others.
+   *
+   * @return     {boolean}  True if access denied, False otherwise.
+   */
+  isAccessDenied(routeName: string): boolean {
+    return !this.$store.state.profileDApp.isMyProfile
+      && this.publicNavEntries.indexOf(routeName) === -1;
```
Comment (karladler) [inline dapps/profile.vue/src/components/root/root.ts]: could also use fancy es6 includes ;)

File: dapps/profile.vue/src/components/root/root.ts [hunk 2]
```diff
@@ -41,6 +41,12 @@ export default class ProfileRootComponent extends mixins(EvanComponent) {
    */
   navEntries: NavEntryInterface[] = [];
 
+  /**
+   * Nav entries that are visible by others
+   */
+  // TODO: add verification to public nav entries, when verifications are public visible
+  publicNavEntries: string[] = ['detail', 'did'];
```
Comment (karladler) [inline dapps/profile.vue/src/components/root/root.ts]: I think the typing would be implicit here, no need to define the type.

Comment (Tschuck) [inline dapps/profile.vue/src/components/root/root.ts]:
```suggestion
  publicNavEntries = ['detail', 'did'];
```

File: dapps/profile.vue/src/components/root/root.ts (hunk 1 above)
Comment (Tschuck) [inline dapps/profile.vue/src/components/root/root.ts]:
```suggestion
      && !this.publicNavEntries.includes(routeName);
```

**Pull request merged**